overlord/snapshotstate/backend/helpers.go

```diff
@@ -0,0 +1,176 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package backend
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+)
+
+func nextTrash(fn string) (string, error) {
+	for n := 1; n < 100; n++ {
+		cand := fmt.Sprintf("%s.~%d~", fn, n)
+		if exists, _, _ := osutil.DirExists(cand); !exists {
```

Comment:
because if somebody looks at their data directory during restore, having it being obviously a backup of an existing directory would be a lot clearer than it being some tempfile thing.

Reply:
We discussed this on the call.. we probably need to do something here on both points. The second one is related to a security issue.